azurerm/resource_arm_app_service_certificate_order.go

```diff
@@ -0,0 +1,488 @@
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          "standard",
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc: validation.StringInSlice([]string{
+					"standard",
+					"wildcard",
+				}, true),
+			},
+
+			"validity_in_years": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 3),
+			},
+
+			"domain_verification_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"serial_number": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"last_certificate_issuance_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"next_auto_renewal_time_stamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"expiration_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"is_private_key_external": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"app_service_certificate_not_renewable_reasons": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"signed_certificate": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: appServiceCertificateDetailsSchema(),
+				},
+			},
+
+			"root": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: appServiceCertificateDetailsSchema(),
+				},
+			},
+
+			"intermediate": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: appServiceCertificateDetailsSchema(),
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmAppServiceCertificateOrderCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).web.CertificatesOrderClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for App Service Certificate creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing App Service Certificate Order %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_app_service_certificate_order", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	t := d.Get("tags").(map[string]interface{})
+
+	// if v, ok := d.GetOk("certificates"); ok { }
+
+	properties := web.AppServiceCertificateOrderProperties{}
+
+	if v, ok := d.GetOk("product_type"); ok {
```

Comment:
We should put every attribute in the properties struct even if Terraform isn't holding it. This allows us to pass empty values to the api to remove certain attributes that aren't present.

Reply:
Get attributes directly and put them into the struct.